test/e2e/rollout_test.go

```diff
@@ -472,6 +473,162 @@ var _ = Describe("placing wrapped resources using a CRP", Ordered, func() {
+					Name:    workNamespace.Name,
+					Version: corev1.SchemeGroupVersion.Version,
+				},
+				{
+					Group:     appv1.SchemeGroupVersion.Group,
+					Version:   appv1.SchemeGroupVersion.Version,
+					Kind:      utils.DeploymentKind,
+					Name:      testDeployment.Name,
+					Namespace: workNamespace.Name,
+				},
+			}
+		})
+
+		It("create the deployment resource in the namespace", func() {
+			Expect(hubClient.Create(ctx, &workNamespace)).To(Succeed(), "Failed to create namespace %s", workNamespace.Name)
+			testDeployment.Namespace = workNamespace.Name
+			Expect(hubClient.Create(ctx, &testDeployment)).To(Succeed(), "Failed to create test deployment %s", testDeployment.Name)
+		})
+
+		It("create the CRP that select the namespace", func() {
+			crp := buildCRPForSafeRollout()
+			crp.Spec.RevisionHistoryLimit = ptr.To(int32(1))
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+		})
+
+		It("should update CRP status as expected", func() {
+			crpStatusUpdatedActual := crpStatusUpdatedActual(wantSelectedResources, allMemberClusterNames, nil, "0")
+			Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP status as expected")
+		})
+
+		It("should place the resources on all member clusters", func() {
+			for idx := range allMemberClusters {
+				memberCluster := allMemberClusters[idx]
+				workResourcesPlacedActual := waitForDeploymentPlacementToReady(memberCluster, &testDeployment)
+				Eventually(workResourcesPlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place work resources on member cluster %s", memberCluster.ClusterName)
+			}
+		})
+
+		It("change the image name in deployment, to make it unavailable", func() {
+			Eventually(func() error {
+				var dep appv1.Deployment
+				err := hubClient.Get(ctx, types.NamespacedName{Name: testDeployment.Name, Namespace: testDeployment.Namespace}, &dep)
+				if err != nil {
+					return err
+				}
+				dep.Spec.Template.Spec.Containers[0].Image = randomImageName
+				return hubClient.Update(ctx, &dep)
+			}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to change the image name in deployment")
+		})
+
+		It("should update CRP status on deployment failed as expected", func() {
+			failedDeploymentResourceIdentifier := placementv1beta1.ResourceIdentifier{
+				Group:     appv1.SchemeGroupVersion.Group,
+				Version:   appv1.SchemeGroupVersion.Version,
+				Kind:      utils.DeploymentKind,
+				Name:      testDeployment.Name,
+				Namespace: testDeployment.Namespace,
+			}
+			crpStatusActual := safeRolloutWorkloadCRPStatusUpdatedActual(wantSelectedResources, failedDeploymentResourceIdentifier, allMemberClusterNames, "1", 2)
+			Eventually(crpStatusActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP status as expected")
+		})
+
+		It("update work to trigger a work generator reconcile", func() {
```

Comment:
Hi Ryan! Is this step necessary? I assume that after the image change the works should already have the false available conditions.

Reply:
At this moment, it seems that the removal of the resource snapshot name part (as in the test description) never happened

Reply:
And normally the work object should have already been marked as unavailable after image change.

Reply:
if the work doesn't change, the already available binding won't get reconcile then its status will stay as "ready" then it won't hit this bug.